### ISSUE TITLE
ArcFace: Additive Angular Margin Loss for Deep Face Recognition

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#2]
# Information
### URL
https://arxiv.org/abs/1801.07698
### Authors
Jiankang Deng, Jia Guo, Niannan Xue, Stefanos Zafeiriou
### Published at
Imperial College London
### Related Websites
[モダンな深層距離学習 (deep metric learning) 手法: SphereFace, CosFace, ArcFace](https://qiita.com/yu4u/items/078054dfb5592cbb80cc)

### Code
[ronghuaiyang/arcface-pytorch](https://github.com/ronghuaiyang/arcface-pytorch)
# Reading Motivation
距離学習を学ぶ. 研究に使えそう.
# Summary
効率的かつ簡単な実装で, ベクトル間の角度にマージンを設定することで直接パラメータを最適化する手法, arcfaceを提案.

# Proposed Method
* softmax loss
![Screenshot from 2021-05-19 04-06-50](https://user-images.githubusercontent.com/39965001/118713046-7ab33900-b85c-11eb-9d30-cf7198693acd.png)
* arcface
角度のみに注目し, 大きさは同一スケール化.  更にマージンを設定し, 正解方向の角度の広がりを小さくする. 
![Screenshot from 2021-05-19 04-05-35](https://user-images.githubusercontent.com/39965001/118713077-7edf5680-b85c-11eb-8eed-5f1ad4a57822.png)

softmax と arcface の違い. どちらも半径が固定されている状況を考えているが, arcface では角度の広がりが小さくなっている.
![Screenshot from 2021-05-19 04-46-06](https://user-images.githubusercontent.com/39965001/118713650-29577980-b85d-11eb-8c46-e14863992a94.png)

学習の流れ
![Screenshot from 2021-05-19 04-48-50](https://user-images.githubusercontent.com/39965001/118714658-9965ff80-b85d-11eb-8c6e-ce49898758e5.png)
